examples/compose/plot_column_transformer.py

```diff
@@ -42,6 +42,51 @@
 from sklearn.svm import LinearSVC
 
 
+_QUOTE_RE = re.compile(r'(writes in|writes:|wrote:|says:|said:'
+                       r'|^In article|^Quoted from|^\||^>)')
+
+
+def strip_newsgroup_quoting(text):
+    """
+    Given text in "news" format, strip lines beginning with the quote
+    characters > or |, plus lines that often introduce a quoted section
+    (for example, because they contain the string 'writes:'.)
+
+    Parameters
+    ----------
+    text : string
+        The text from which to remove the signature block.
+    """
+    good_lines = [line for line in text.split('\n')
+                  if not _QUOTE_RE.search(line)]
+    return '\n'.join(good_lines)
+
+
+def strip_newsgroup_footer(text):
+    """
+    Given text in "news" format, attempt to remove a signature block.
+
+    As a rough heuristic, we assume that signatures are set apart by either
+    a blank line or a line made of hyphens, and that it is the last such line
+    in the file (disregarding blank lines at the end).
+
+    Parameters
+    ----------
+    text : string
+        The text from which to remove the signature block.
+    """
+    lines = text.strip().split('\n')
+    for line_num in range(len(lines) - 1, -1, -1):
+        line = lines[line_num]
+        if line.strip().strip('-') == '':
+            break
+
+    if line_num > 0:
+        return '\n'.join(lines[:line_num])
+    else:
+        return text
```

Comment:
I just realized instead of having these here, you can pass `remove=('footer', 'quoting')` to `fetch_20newsgroups` down in the bottom of the file, and remove these from the example, and simplify the `SubjectBodyExtractor` as well.

Reply:
PR updated to reflect this change.